### PR TITLE
Enable TEST_MULTIGPU unit tests which pass with ROCm 2.2 on Vega20

### PR DESCRIPTION
**Pull request by jithunnair-amd**

### CODE REVIEW

File: test/test_nn.py [hunk 1]
```diff
@@ -3183,7 +3182,6 @@ def _test_scatter(self, tensor):
         _assertGradAndGradgradChecks(self, lambda y: dp.scatter(y, (0, 1)), (x,))
 
     @unittest.skipIf(not TEST_MULTIGPU, "multi-GPU not supported")
-    @skipIfRocm
```
Comment (iotamudelta) [inline test/test_nn.py]: Leave in for now.

File: test/test_nn.py [hunk 2]
```diff
@@ -3234,7 +3232,6 @@ def _test_gather(self, output_device):
         _assertGradAndGradgradChecks(self, lambda x, y: dp.gather((x, y), output_device), inputs)
 
     @unittest.skipIf(not TEST_MULTIGPU, "multi-GPU not supported")
-    @skipIfRocm
```
Comment (iotamudelta) [inline test/test_nn.py]: Leave in for now.

**Comment by jithunnair-amd** [comment]
What's the flaky error we see for these two tests: test_scatter_cpu and test_gather_cpu?

**Comment by iotamudelta** [comment]
Wrong results, IIRC

**Comment by jithunnair-amd** [comment]
Just to see if there's any flakiness
@pytorchbot retest this please

**Comment by iotamudelta** [comment]
@pytorchbot retest this please

after NCCL disablement

**Comment by jithunnair-amd** [comment]
Cool, seems to work.

**Pull request merged**